### PR TITLE
Feature/nijo 931 documentation and booking structure simplification

### PR DESCRIPTION
**Pull request by kirillta**

### CODE REVIEW

File: HappyTravel.EdoContracts/Accommodations/Accommodation.cs [hunk 1]
```diff
@@ -95,7 +95,7 @@ namespace HappyTravel.EdoContracts.Accommodations
         public string? HotelChain { get; }
 
         /// <summary>
-        ///     The Happytravel.com unique accommodation ID.
+        ///     The Happytravel.com unique accommodation correlation ID.
```
Comment (mingazhev) [inline HappyTravel.EdoContracts/Accommodations/Accommodation.cs:98]: Why correlation?

File: HappyTravel.EdoContracts/Accommodations/Internals/CancellationPolicy.cs [hunk 2]
```diff
@@ -8,23 +8,30 @@ namespace HappyTravel.EdoContracts.Accommodations.Internals
     public readonly struct CancellationPolicy
     {
         [JsonConstructor]
-        public CancellationPolicy(DateTime fromDate, double percentage)
+        public CancellationPolicy(in DateTime fromDate, double percentage)
```
Comment (mingazhev) [inline HappyTravel.EdoContracts/Accommodations/Internals/CancellationPolicy.cs:11]: Not sure that it is effective for dates

**Review by mingazhev — APPROVED** [review]
With comments

File: HappyTravel.EdoContracts/Accommodations/Enums/LocationDescriptionCodes.cs [hunk 3]
```diff
@@ -7,21 +8,35 @@ namespace HappyTravel.EdoContracts.Accommodations.Enums
     public enum LocationDescriptionCodes
     {
         Unspecified = 0,
+
+        [Description("City center")]
```
Comment (hasmikharutyunyan) [inline HappyTravel.EdoContracts/Accommodations/Enums/LocationDescriptionCodes.cs:12]: Why not a comment, but an attribute?

File: HappyTravel.EdoContracts/Accommodations/Internals/RoomContract.cs [hunk 4]
```diff
@@ -32,31 +30,54 @@ namespace HappyTravel.EdoContracts.Accommodations.Internals
...
+
+        /// <summary>
+        ///     The numerical code of a contract type.
+        /// </summary>
+        public int ContractTypeCode { get; }
+
         public bool IsAvailableImmediately { get; }
+
+        /// <summary>
+        ///     Indicates if a contract a dynamic offer.
```
Comment (hasmikharutyunyan) [inline HappyTravel.EdoContracts/Accommodations/Internals/RoomContract.cs:64]: Could we here describe what mean dynamic, or it's better do in documentations?

File: HappyTravel.EdoContracts/Accommodations/Internals/SlimAccommodation.cs [hunk 5]
```diff
@@ -8,41 +8,72 @@ namespace HappyTravel.EdoContracts.Accommodations.Internals
...
 
 
         public override bool Equals(object? obj) => obj is SlimAccommodation other && Equals(other);
 
 
-        public bool Equals(SlimAccommodation other)
-            => (Id, Location, Name, Picture, Rating, PropertyType)
-                .Equals((other.Id, other.Location, other.Name, other.Picture, other.Rating, other.PropertyType));
+        public bool Equals(in SlimAccommodation other)
+            => (Id, Location, Name, Picture: Photo, Rating, PropertyType)
```
Comment (hasmikharutyunyan) [inline HappyTravel.EdoContracts/Accommodations/Internals/SlimAccommodation.cs:71]: It is previous naming

File: HappyTravel.EdoContracts/Accommodations/Internals/RoomContract.cs (hunk 4 above)
Comment (kirillta) [reply HappyTravel.EdoContracts/Accommodations/Internals/RoomContract.cs:64]: I'll do that in the documentation

File: HappyTravel.EdoContracts/Accommodations/Enums/LocationDescriptionCodes.cs (hunk 3 above)
Comment (kirillta) [reply HappyTravel.EdoContracts/Accommodations/Enums/LocationDescriptionCodes.cs:12]: To use textual forms of these descriptions in emails etc.

File: HappyTravel.EdoContracts/Accommodations/Internals/CancellationPolicy.cs (hunk 2 above)
Comment (kirillta) [reply HappyTravel.EdoContracts/Accommodations/Internals/CancellationPolicy.cs:11]: Yes, you're right

File: HappyTravel.EdoContracts/Accommodations/Accommodation.cs (hunk 1 above)
Comment (kirillta) [reply HappyTravel.EdoContracts/Accommodations/Accommodation.cs:98]: It's the nature of HtId. Do you think we need to remove the word from the comment?

**Review by hasmikharutyunyan — APPROVED** [review]

**Pull request merged**